process-document/src/main/kotlin/com/ritense/processdocument/resolver/DocumentTableValueResolver.kt

```diff
@@ -82,10 +82,10 @@ class DocumentTableValueResolver(
                 "assigneeId" -> document.assigneeId()
                 "createdBy" -> document.createdBy()
                 "createdOn" -> document.createdOn()
-                "definitionId" -> document.definitionId()
-                "definitionId.name" -> document.definitionId().name()
-                // TODO: change definitionId.caseDefinitionId
-                "definitionId.version" -> document.definitionId().caseDefinitionId()
+                "documentDefinitionId" -> document.definitionId()
+                "documentDefinitionId.name" -> document.definitionId().name()
+                "definitionId.key" -> document.definitionId().caseDefinitionId().key
+                "definitionId.versionTag" -> document.definitionId().caseDefinitionId().versionTag.version
```

Comment:
Two different domains are now being mixed in a single value resolver. CaseDefinition and DocumentDefinition. It would be better to keep them separate

Reply:
I agree it would be nicer to fully separate the two, but since a case definition and document definition are 1:1, I think it's fine here.

Reply:
Not sure i agree. this isn't about getting the information of the document definition, it's about getting the information about the document. This includes relations like of what type the document is, and what the related case definition is.

Reply:
Fair, then there has been a problem with this class since it was conceived, as it has had the `case` prefix since the beginning. This is misleading, as the data we're working with is document based.

On that note, this shows there's a flaw with the value resolvers. There can only ever be a single value resolver per prefix. But we see a big overlap in what data `DocumentJsonValueResolverFactory` and the `DocumentTableValueResolver` do. Where they differ is how they handle their values, and this is problematic since it would make sense for both of them to use the `doc` prefix.